src/software/ai/hl/stp/tactic/get_possession/get_possession_fsm.h

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include "shared/constants.h"
+#include "software/ai/evaluation/pass.h"
+#include "software/ai/hl/stp/tactic/move/move_fsm.h"
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/intent/move_intent.h"
+#include "software/geom/algorithms/acute_angle.h"
+#include "software/geom/algorithms/contains.h"
+#include "software/geom/algorithms/distance.h"
+
+struct GetPossessionFSM
+{
+    class InterceptBallState;
+
+    struct ControlParams
+    {
+    };
+
+    DEFINE_UPDATE_STRUCT_WITH_CONTROL_AND_COMMON_PARAMS
+
+    /**
+     * Calculates the interception point for intercepting balls
+     *
+     * @param robot The robot to do the interception
+     * @param ball The ball to intercept
+     * @field The field to intercept on
+     *
+     * @return the best interception point
+     */
+    // TODO (#1968): Merge this functionality with findBestInterceptForBall in the
+    // evaluation folder
+    static Point findInterceptionPoint(const Robot &robot, const Ball &ball,
+                                       const Field &field)
+    {
+        static constexpr double BALL_MOVING_SLOW_SPEED_THRESHOLD   = 0.3;
+        static constexpr double INTERCEPT_POSITION_SEARCH_INTERVAL = 0.1;
+        if (ball.velocity().length() < BALL_MOVING_SLOW_SPEED_THRESHOLD)
+        {
+            auto face_ball_vector = (ball.position() - robot.position());
+            auto point_in_front_of_ball =
+                ball.position() -
+                face_ball_vector.normalize(DIST_TO_FRONT_OF_ROBOT_METERS +
+                                           BALL_MAX_RADIUS_METERS);
+            return point_in_front_of_ball;
+        }
+        Point intercept_position = ball.position();
+        while (contains(field.fieldLines(), intercept_position))
```

Comment:
I remember I wrote something similar to this a long time ago, but has anyone tried to make a closed-form solution to this?

Reply:
Yeah that's a bit outside of the scope of this PR since I'm mainly trying to port over existing behaviour. #1968 tracks improvements to how we do intercept